plugins/rss-feeds/src/main.tsx

```diff
@@ -0,0 +1,40 @@
+import "./globals.css"
+
+import { framer } from "framer-plugin"
+import React from "react"
+import ReactDOM from "react-dom/client"
+import { App, importData } from "./App.tsx"
+
+async function runPlugin() {
+  const mode = await framer.getMode()
+  const collection = await framer.getCollection()
+
+  const rssSourceId = await collection.getPluginData("rssSourceId")
+  if (mode === "syncCollection" && rssSourceId) {
+    try {
+      await importData(collection, rssSourceId)
+      await framer.closePlugin()
+    } catch {
+      // Failed to synchronize - show configuration
+    }
+  }
```

Comment:
Why is this logic in this file? I find it nice when everything is inside the App.tsx for these smaller examples

Reply:
When a plugin is launched in `syncCollection` mode we can synchronize the CMS without displaying any UI.

The reason this is placed here is that we check if we can synchronize the data without displaying any UI.

If it is possible we skip mounting the React app altogether and close the plugin after syncing 🙏

Reply:
ok makes sense! lets keep it then